agent/lib/kontena/workers/volumes/volume_manager.rb

```diff
@@ -124,6 +125,14 @@ def volume_exist?(volume_name, driver)
       end
     end
 
+    def drivers_match?(expected, actual)
+      if expected.include?(':')
+        expected == actual
+      else
+        expected == actual.split(':', 2)[0]
```

Comment:
One optional for expected Kontena volume drivers without a version suffix would be to check that the actual volume driver is either a built-in like `local` without any version suffix, or a plugin with the `latest` version suffix.